mlflow/genai/utils/trace_utils.py

```diff
@@ -419,6 +424,65 @@ def _parse_chunk(chunk: Any) -> dict[str, Any] | None:
     return doc
 
 
+def clean_up_extra_traces(run_id: str, start_time_ms: int) -> None:
+    """
+    Clean up noisy traces generated outside predict function.
+
+    Evaluation run should only contain traces that is being evaluated or generated by the predict
+    function. If not, the result will not show the correct list of traces.
+    Sometimes, there are extra traces generated during the evaluation, for example, custom scorer
+    code might generate traces. This function cleans up those noisy traces.
+
+    Args:
+        run_id: The ID of the run to clean up.
+        start_time_ms: The start time of the evaluation in milliseconds.
+    """
+    from mlflow.tracking.fluent import _get_experiment_id
+
+    try:
+        # Search for all traces generated during evaluation
+        traces = mlflow.search_traces(
+            run_id=run_id,
+            include_spans=False,
+            filter_string=f"trace.timestamp >= {start_time_ms}",
+            return_type="list",
+        )
+
+        extra_trace_ids = [trace.info.trace_id for trace in traces if not _should_keep_trace(trace)]
+        if extra_trace_ids:
+            _logger.debug(
+                f"Found {len(extra_trace_ids)} extra traces generated during evaluation run. "
+                "Deleting them."
+            )
+            # Import MlflowClient locally to avoid issues with tracing-only SDK
+            from mlflow.tracking.client import MlflowClient
+
+            MlflowClient().delete_traces(
+                experiment_id=_get_experiment_id(), trace_ids=extra_trace_ids
+            )
+            for trace_id in extra_trace_ids:
+                IPythonTraceDisplayHandler.get_instance().traces_to_display.pop(trace_id, None)
+        else:
+            _logger.debug("No extra traces found during evaluation run.")
+    except Exception as e:
+        _logger.warning(
```

Comment:
I wonder do we need this to be warning? Probably debug since users may just feel this annoying